attachment_preview/static/src/js/components/chatter/chatter.esm.js

```diff
@@ -0,0 +1,272 @@
+                        };
+                    })
+                );
+
+                rpc.query({
+                    model: "ir.attachment",
+                    method: "get_attachment_extension",
+                    args: [
+                        _.map(_.keys(attachments), function (id) {
+                            return parseInt(id, 10);
+                        }),
+                    ],
+                }).then(
+                    function (extensions) {
+                        var reviewableAttachments = _.map(
+                            _.keys(
+                                _.pick(extensions, function (extension) {
+                                    return canPreview(extension);
+                                })
+                            ),
+                            function (id) {
+                                return {
+                                    id: id,
+                                    url: attachments[id].url,
+                                    extension: extensions[id],
+                                    title: attachments[id].title,
+                                    previewUrl: getUrl(
+                                        id,
+                                        attachments[id].url,
+                                        extensions[id],
+                                        attachments[id].title
+                                    ),
+                                };
+                            }
+                        );
+
+                        deferred.resolve(reviewableAttachments);
+                    },
+
+                    function () {
+                        deferred.reject();
+                    }
+                );
+
+                return deferred.promise();
+            },
+
+            _updatePreviewButtons: function (previewableAttachments) {
+                $(this)
+                    .find(".o_attachment_preview")
+                    .each(function () {
+                        var $this = $(this);
+                        var att = _.findWhere(previewableAttachments, {
+                            id: $this.attr("data-id"),
+                        });
+                        if (att) {
+                            $this.attr("data-extension", att.extension);
+                        } else {
+                            $this.remove();
+                        }
+                    });
+            },
+        }
+    );
+
+    basic_fields.FieldBinaryFile.include(components.Chatter);
+    basic_fields.FieldBinaryFile.include({
+        showPreview(
+            attachment_id,
+            attachment_url,
+            attachment_extension,
+            attachment_title,
+            split_screen
+        ) {
+            var url = getUrl(
+                attachment_id,
+                attachment_url,
+                attachment_extension,
+                attachment_title
+            );
```

Comment:
I check if the extensions can be preview should be added
```
                if (!canPreview(attachment_extension))
                {
                    return;
                }
```

Reply:
it's added!